app/models/transaction.rb

```diff
@@ -0,0 +1,145 @@
+# frozen_string_literal: true
+
+class Transaction < ApplicationRecord
+  has_many :histories, dependent: :destroy, class_name: 'TransactionHistory'
+
+  monetize :amount_cents, allow_nil: false, numericality: { greater_than: 0 }
+
+  enum gateway_name: Constants::GATEWAY_NAMES
+  enum amount_currency: Constants::CURRENCY_LIST
+
+  validates :order_id, format: { with: Constants::ORDER_ID_REGEX }
+
+  default_scope { includes(:histories) }
+
+  def purchase!(credit_card, customer)
+    transition!(from: :pending, to: :captured) do
+      gateway.purchase(
+        gateway_amaount,
```

Comment:
Just a little typo here.
```suggestion
        gateway_amount,
```